src/worker/ingestion/process-event.ts

```diff
@@ -607,4 +604,21 @@ export class EventsProcessor {
         }
         return data
     }
+
+    private async createPersonIfDistinctIdIsNew(
+        teamId: number,
+        distinctId: string,
+        sentAt: DateTime,
+        personUuid: string
+    ): Promise<void> {
+        const isNewPerson = await this.personManager.isNewPerson(this.db, teamId, distinctId)
+        if (isNewPerson) {
+            // Catch race condition where in between getting and creating, another request already created this user
+            try {
+                await this.db.createPerson(sentAt, {}, teamId, null, false, personUuid.toString(), [distinctId])
+            } catch (error) {
+                Sentry.captureException(error, { extra: { teamId, distinctId, sentAt, personUuid } })
```

Comment:
Is this worth notifying sentry of?

Reply:
It's probably a bit of noise, but I'd like to have at least a bit of context on Sentry about all these race conditions we handle. This context could be helpful the next time we have a race condition issue to debug.

Reply:
Ack. Worth keeping this in mind though since sentry does have an error rate limit, very frequent errors will cause us other issues :)